contracts/interchain-token-service/src/contract/migrations/mod.rs

```diff
@@ -1,16 +1,442 @@
-use axelar_wasm_std::migrate_from_version;
+use std::collections::HashMap;
+
+use axelar_wasm_std::address::validate_cosmwasm_address;
+use axelar_wasm_std::{migrate_from_version, IntoContractError};
+use cosmwasm_schema::cw_serde;
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{DepsMut, Empty, Env, Response};
+use cosmwasm_std::{DepsMut, Env, Response, Storage};
+use cw_storage_plus::Map;
+use error_stack::ResultExt;
+use router_api::{Address, ChainNameRaw};
+
+use crate::shared::NumBits;
+use crate::state::{save_chain_config, ChainConfig as NewChainConfig};
+
+// Old state structure (before migration)
+#[cw_serde]
+pub struct OldTruncationConfig {
+    pub max_uint_bits: NumBits,
+    pub max_decimals_when_truncating: u8,
+}
 
-pub type MigrateMsg = Empty;
+#[cw_serde]
+pub struct OldChainConfig {
+    pub truncation: OldTruncationConfig,
+    pub its_address: Address,
+    pub frozen: bool,
+    // Note: no translation_contract field in old state
+}
+
+// Storage map for old chain configs
+const OLD_CHAIN_CONFIGS: Map<&ChainNameRaw, OldChainConfig> = Map::new("chain_configs");
+
+#[cw_serde]
+pub struct ChainTranslationConfig {
+    pub chain: ChainNameRaw,
+    pub translation_contract: Address,
+}
+
+#[cw_serde]
+pub struct MigrateMsg {
+    /// List of chain and translation contract address pairs
+    pub chain_translation_configs: Vec<ChainTranslationConfig>,
```

Comment:
Since all chains use ABI translator at the moment, the migrate msg should specify a single abi_translator contract to use for all chains. It can then be upgraded via UpdateConfig msg if a chain decides to switch